shell/platform/linux/fl_renderer_x11.cc

```diff
@@ -25,9 +25,12 @@ static void fl_renderer_x11_class_init(FlRendererX11Class* klass) {
 
 static void fl_renderer_x11_init(FlRendererX11* self) {}
 
-FlRendererX11* fl_renderer_x11_new(Window xid) {
-  FlRendererX11* self = static_cast<FlRendererX11*>(
+FlRendererX11* fl_renderer_x11_new() {
+  return static_cast<FlRendererX11*>(
```

Comment:
It looks like some unrelated changes ended up in this PR?

Reply:
I had to change the renderer to set the X11 window ID after construction - we need the engine as soon as the FlView is created, so we can register any platform message callbacks. However we don't know the X11 window ID until later. So the FlRendererX11 needed updating for that. It doesn't feel great, as it would be nicer for everything to be set at construction. We could resolve this by making a new object for sending and receiving messages, but that makes the API user do more work for no apparent gain. I ended up going with this solution for now and consider changing it later if that new class ends up being useful.